### PR TITLE
Add jakartaredhat WildFly Core Profile tck runner source

### PR DESCRIPTION
**Pull request by scottmarlow**
Merge of https://github.com/jakartaredhat/wf-core-tck-runner commits

Followed advice from https://stackoverflow.com/questions/37471740/how-to-copy-commits-from-one-git-repo-to-another (been a while since I did this):

git clone https://github.com/wildfly/wildfly-tck-runners.git new-repo
cd new-repo
git remote add old https://github.com/jakartaredhat/wf-core-tck-runner.git
git remote update
git log old/main
last-commit-hash-from-old-repo = ad73e6919a35e9d8e56352e1382fe47489f78abe
git merge --allow-unrelated-histories <last-commit-hash-from-old-repo>
or git merge --allow-unrelated-histories ad73e6919a35e9d8e56352e1382fe47489f78abe
git push origin main

### CODE REVIEW

File: wf-core-tck-runner/README-WFP.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
+= WildFly Preview Notes
```
Comment (bstansberry) [inline wf-core-tck-runner/README-WFP.adoc:1]: LATER: We should update this page to focus on standard WildFly.

File: wf-core-tck-runner/cdi-tck/pom.xml [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+    <dependency>
+      <groupId>org.jboss.weld</groupId>
+      <artifactId>weld-porting-package-tck</artifactId>
+      <version>${weld.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>jakarta.ee.tck.coreprofile</groupId>
+      <artifactId>cdi-lite-tck-suite</artifactId>
+      <version>10.0.0</version>
```
Comment (bstansberry) [inline wf-core-tck-runner/cdi-tck/pom.xml:89]: LATER: Shouldn't this be ${core.profile.tck.version} ?

File: wf-core-tck-runner/cdi-tck/pom.xml [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+    <dependency>
+      <groupId>org.jboss.arquillian.testng</groupId>
+      <artifactId>arquillian-testng-container</artifactId>
+      <version>${arquillian.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.jboss.shrinkwrap.descriptors</groupId>
+      <artifactId>shrinkwrap-descriptors-impl-javaee</artifactId>
+      <version>2.0.0</version>
```
Comment (bstansberry) [inline wf-core-tck-runner/cdi-tck/pom.xml:124]: LATER: Consider controlling this version in a property.

File: wf-core-tck-runner/jsonp-tck/pom.xml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  <dependencies>
+
+    <dependency>
+      <groupId>jakarta.json</groupId>
+      <artifactId>jakarta.json-tck-tests</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>jakarta.ee.tck.coreprofile</groupId>
+      <artifactId>core-tck-jsonp-extension</artifactId>
+      <version>10.0.0-SNAPSHOT</version>
```
Comment (bstansberry) [inline wf-core-tck-runner/jsonp-tck/pom.xml:36]: LATER: This is odd

File: wf-core-tck-runner/pom.xml [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+  </developers>
+
+  <organization>
+    <name>Jakarta Core Profile TCK Runner (10.0) for WildFly/Weld</name>
+    <url>https://github.com/jakartaredhat/wf-core-tck-runner</url>
+  </organization>
+
+  <issueManagement>
+    <system>GitHub</system>
+    <url>https://github.com/jakartaredhat/wf-core-tck-runner/issues</url>
```
Comment (bstansberry) [inline wf-core-tck-runner/pom.xml:52]: This should be updated.

File: wf-core-tck-runner/rest-tck/pom.xml [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>jakarta.ee.tck.coreprofile</groupId>
+      <artifactId>rest-tck-suite</artifactId>
+      <version>10.0.0</version>
```
Comment (bstansberry) [inline wf-core-tck-runner/rest-tck/pom.xml:44]: LATER: Use a property

File: wf-core-tck-runner/rest-tck/pom.xml [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+
+          <argLine>
+            --add-exports=java.desktop/sun.awt=ALL-UNNAMED
+            --add-opens=java.base/java.io=ALL-UNNAMED
+            --add-opens=java.base/java.lang=ALL-UNNAMED
+            --add-opens=java.base/java.lang.invoke=ALL-UNNAMED
+            --add-opens=java.base/java.security=ALL-UNNAMED
+            --add-opens=java.base/java.util=ALL-UNNAMED
+            --add-opens=java.management/javax.management=ALL-UNNAMED
+            --add-opens=java.naming/javax.naming=ALL-UNNAMED
```
Comment (bstansberry) [inline wf-core-tck-runner/rest-tck/pom.xml:210]: LATER: Create a property in the root pom and re-use it.

File: wf-core-tck-runner/run_all.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  exit 1
+fi
+
+# Check
+ca_jar_size=($(wc -c  ${JBOSS_HOME}/modules/system/layers/base/jakarta/annotation/api/main/jakarta.annotation-api-2.1.0.jar ))
+if [[ ${ca_jar_size} != 26141 ]]; then
+  echo "Patching WFP jakarta.annotation-api-2.1.0.jar to 2.1.1..."
+  wget -O ${JBOSS_HOME}/modules/system/layers/base/jakarta/annotation/api/main/jakarta.annotation-api-2.1.0.jar \
+    https://repo1.maven.org/maven2/jakarta/annotation/jakarta.annotation-api/2.1.1/jakarta.annotation-api-2.1.1.jar
+fi
```
Comment (bstansberry) [inline wf-core-tck-runner/run_all.sh:29]: LATER: Remove this, and replace all other refs in wf-core-tck-runner with 2.1.1. That's what we use in WF.

File: wf-core-tck-runner/cdi-tck/pom.xml [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+
+    <profile>
+      <id>cdi-signature-test</id>
+      <!-- API jars from the WildFly distribution -->
+      <dependencies>
+        <dependency>
+          <groupId>jakarta.annotation</groupId>
+          <artifactId>jakarta.annotation-api</artifactId>
+          <scope>system</scope>
+          <systemPath>${env.JBOSS_HOME}/modules/system/layers/base/jakarta/annotation/api/main/jakarta.annotation-api-2.1.0.jar</systemPath>
```
Comment (bstansberry) [inline wf-core-tck-runner/cdi-tck/pom.xml:374]: LATER: Use properties for all these versions.

**Review by bstansberry — APPROVED** [review]
Thanks @scottmarlow 

This is good as is. I made a bunch of comments starting with "LATER" but those are meant as a list of future work for this runner, not a request for changes in this PR. Reviewing the PR was just a convenient time to note things.

There's one comment not labeled "LATER" re the scm url in the root pom. That's related to this task of code migration, but it's fine IMHO to do that later as well.

File: wf-core-tck-runner/README-WFP.adoc (hunk 1 above)
Comment (scottmarlow) [reply wf-core-tck-runner/README-WFP.adoc]: Addressed

File: wf-core-tck-runner/jsonp-tck/pom.xml (hunk 4 above)
Comment (scottmarlow) [reply wf-core-tck-runner/jsonp-tck/pom.xml]: updated jsonp-tck/pom.xml to not reference the `10.0.0-SNAPSHOT` as this does look wrong (switched to ${core.profile.tck.version}.

File: wf-core-tck-runner/pom.xml (hunk 5 above)
Comment (scottmarlow) [reply wf-core-tck-runner/pom.xml]: Updated

File: wf-core-tck-runner/run_all.sh (hunk 8 above)
Comment (scottmarlow) [reply wf-core-tck-runner/run_all.sh]: done

File: wf-core-tck-runner/cdi-tck/pom.xml (hunk 9 above)
Comment (scottmarlow) [reply wf-core-tck-runner/cdi-tck/pom.xml]: :+1: 

Perhaps we could set the property based on something like doing the following in the WildFly folder:
annotationapi=`find -name jakarta*annotation*api*.jar`

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**